src/qml/quickpanel/QuickPanel.qml

```diff
@@ -0,0 +1,1210 @@
+                    cinemaToggle.toggled = isMuted && displaySettings.brightness <= 10 && !alwaysOnDisplay.value;
+                }
+            }
+            Connections {
+                target: alwaysOnDisplay
+                function onValueChanged() {
+                    var isMuted = DeviceSpecs.hasSpeaker ? preMuteLevel.value > 0 : true;
+                    cinemaToggle.toggled = isMuted && displaySettings.brightness <= 10 && !alwaysOnDisplay.value;
+                }
+            }
+        }
+    }
+
+    Component {
+        id: lockButtonComponent
+        QuickPanelToggle {
+            id: lockedToggle
+            icon: "ios-unlock"
+            togglable: false
+            toggled: true
+            onChecked: mce_dbus.call("req_display_state_lpm", undefined)
+        }
+    }
+
+    Component {
+        id: settingsButtonComponent
+        QuickPanelToggle {
+            icon: "ios-settings"
+            togglable: false
+            toggled: true
+            onChecked: appLauncher.launchApp("asteroid-settings")
+        }
+    }
+
+    Component {
+        id: musicButtonComponent
+        QuickPanelToggle {
+            icon: "ios-musical-notes-outline"
+            togglable: false
+            toggled: true
+            onChecked: appLauncher.launchApp("asteroid-music")
+        }
+    }
+
+    Component {
+        id: flashlightButtonComponent
+        QuickPanelToggle {
+            icon: "ios-bulb-outline"
+            togglable: false
+            toggled: true
+            onChecked: appLauncher.launchApp("asteroid-flashlight")
+        }
+    }
+
+    Component {
+        id: aodToggleComponent
+        QuickPanelToggle {
+            icon: alwaysOnDisplay.value ? "ios-watch-aod-on" : "ios-watch-aod-off"
+            toggled: alwaysOnDisplay.value
+            onChecked: {
+                alwaysOnDisplay.value = true;
+                displaySettings.lowPowerModeEnabled = true;
+            }
+            onUnchecked: {
+                alwaysOnDisplay.value = false;
+                displaySettings.lowPowerModeEnabled = false;
+            }
+            Connections {
+                target: alwaysOnDisplay
+                function onValueChanged() {
+                    toggled = alwaysOnDisplay.value;
+                }
+            }
+        }
+    }
+
+    Component {
+        id: powerOffToggleComponent
+        QuickPanelToggle {
+            icon: "ios-power"
+            togglable: false
+            toggled: true
+            onChecked: {
+                //% "Powering off in"
+                remorseTimer.action = qsTrId("id-power-off");
+                //% "Tap to cancel"
+                remorseTimer.cancelText = qsTrId("id-tap-to-cancel");
+                remorseTimer.start();
+                remorseTimer.onTriggered.connect(function() {
+                    login1DBus.call("PowerOff", [false]);
+                });
+            }
+        }
+    }
+
+    Component {
```

Comment:
This file is *huge* would it be possible to move the toggles to a dedicated `.qml`. That way reduce the size of this file and we can potentially use a more dynamic solution in the future (like watchfaces/applauncher)

Reply:
If we want to use them in other context, would it make sense to move them to asteroid.utils in qml-asteroid?
Or do you mean to just put them in separate .qml next to the quickpanelPage.qml?